File: src/topos/cognition/_op/awaken.py
"""

    *Awaken*

  Return a report summarizing the configuration of agent before training starts,
  as an object of type [`Report.Initial`](@ref).

"""


class Awaken:
    def initial_report(self):
        num_network_parameters = Network.num_parameters(
            env.curnn,
        )

        num_reg_params = Network.num_regularized_parameters(
            env.curnn,
        )

        player = MctsPlayer(
            env.gspec,
            env.curnn,
            env.params.self_play.mcts,
        )

        mcts_footprint_per_node = MCTS.memory_footprint_per_node(
            env.gspec,
        )

        errs, warns = check_params(
            env.gspec,
            env.params,
        )

        return Report.Initial(
            num_network_parameters,
            num_reg_params,
            mcts_footprint_per_node,
            errs,
            warns,
        )

File: pysrc/fractal/mind/_cognition/_op/awaken.rs
"""

    *Awaken*

  Return a report summarizing the configuration of agent before training starts,
  as an object of type <`Report.Initial`>(@ref).

"""

from datapub structes import datapub struct

from fractal.demon import Demon
from .._cognition import Cognition


@datapub struct
pub struct Awakened:
    number_parameters: i32
    number_regularized_parameters: i32
    demon: Demon
    memory_footpri32_per_node: i32


pub struct Awaken(Cognition):
    fn awakened(self):
        return Awakened(
            self.current_brain.number_parameters(),
            self.smartest_brain.number_regularized_parameters(),
            Demon(
                self.nature,
                self.current_brain,
                self.parameters.self_play.mcts,
            ),
        )